### ISSUE TITLE
Nav-bar consistency error.

### ISSUE THREAD

**Issue by MrJoy** [shakacode/react-webpack-rails-tutorial#413]
When going to https://www.reactrails.com/ and clicking `Test React Router ('/react-router')`, the navigation bar entry for `React Router Demo` remains selected.  When loading https://www.reactrails.com/react-router directly, it is not.

I'm not sure which behavior is the intended one, but the inconsistency suggests a bug.

**Comment by justin808** [comment]
@MrJoy I agree. Can you take a look and let me know what you find?

**Comment by justin808** [comment]
If anybody sees this issue, it's a great way to try adding a contribution to this project.

**Comment by justin808** [comment]
@bluemihai did you want to submit a PR?

**Comment by ramrudra** [comment]
@justin808 What is the intended behaviour here?

**Current:**
Test React Router ('/react-router'), the navigation bar entry for React Router Demo remains selected.

Is this the intended behaviour?
 **React Router Demo** menu item should be active when visiting '/react-router'

**Comment by justin808** [comment]
@ramrudra 

* when you click the link for the home page (/) and then go to click the link for Test React Router ('/react-router'), you see the highlight of react-router stays
* but not if you go to the second directly, say from a bookmark

**Comment by mdiaz00147** [comment]
@justin808 I'd like to try on this one. May I? I'm applying for upwork position as well thx

**Comment by tahsin352** [comment]
@justin808 in react router 4, first, need to import withRouter like: import {withRouter} from 'react-router-dom';  then we can get the current path this.props.location.pathname

we can match it and then add 'active' class to respective menu item. thx.

**Comment by justin808** [comment]
Please link to a PR if you want to suggest some fix.

I'm more eager to get the webpack configuration updated and make a sensible configuration for hot reloading.

**Comment by Sirbuland** [comment]
@justin808 
Looks like it is working fine.

<img width="1278" alt="screenshot 2019-02-22 at 7 10 30 pm" src="https://user-images.githubusercontent.com/23502541/53247993-7a0bb680-36d6-11e9-8511-745ed06373ec.png">

**Comment by tahsin352** [comment]
> Please link to a PR if you want to suggest some fix.
> 
> I'm more eager to get the webpack configuration updated and make a sensible configuration for hot reloading.

@justin808 sorry, i might be wrong, but it i guess it is problem with routing, it has nothing with hot-reloading, coz, hot-reloading is a solution to display any update made to any component on the page. but here we are fixing the routing issue. we can use NavLink from react-dom-router. you can see this example for reference : https://codeburst.io/getting-started-with-react-router-5c978f70df91 . it can be done with webpack also. thx.

**Comment by dreamsoftech** [comment]
@justin808 When I debug to fix this issue, I found some interesting thing. There are duplicated providers.
![image](https://user-images.githubusercontent.com/4368899/56544729-e1939480-65a7-11e9-9240-d15ff5260b44.png)
To fix this issue, I tried to remove NavigationBarApp in [clientRegistration.jsx](https://github.com/dreamsoftech/react-webpack-rails-tutorial/blob/e2e3054dc19f4ecfa630271372de0e8c024d875b/client/app/bundles/comments/startup/clientRegistration.jsx#L26).
It fixed this issue but replicates console error. Since I am not familiar with this code, I need to discuss something with you.
